pkg/utils/kubernetes/kubernetes.go

```
@@ -77,6 +78,67 @@ func HasMetaDataAnnotation(meta metav1.Object, key, value string) bool {
 	return ok && val == value
 }
 
+// SetAnnotationsAndUpdate sets the given annotations on the given object and updates it, taking care of conflicts.
+func SetAnnotationsAndUpdate(ctx context.Context, c client.Client, obj client.Object, keyValues ...string) error {
+	annotations := keyValuesToAnnotations(keyValues...)
+	if !hasAnnotations(obj, annotations) {
+		return clientretry.RetryOnConflict(clientretry.DefaultBackoff, func() error {
+			setAnnotations(obj, annotations)
+			return c.Update(ctx, obj)
+		})
+	}
+	return nil
+}
+
+// DeleteAnnotationsAndUpdate deletes the given annotations on the given object and updates it, taking care of conflicts.
+func DeleteAnnotationsAndUpdate(ctx context.Context, c client.Client, obj client.Object, keys ...string) error {
+	if !doesNotHaveAnnotationKeys(obj, keys...) {
+		return clientretry.RetryOnConflict(clientretry.DefaultBackoff, func() error {
+			deleteAnnotations(obj, keys...)
+			return c.Update(ctx, obj)
+		})
+	}
+	return nil
+}
+
+func keyValuesToAnnotations(keyValues ...string) map[string]string {
```

This func will silently ignore the last key, if the value is missing (i.e. odd number of args).
I generally don't like such functions, that will silently swallow programming level mistakes, that can be avoided by using proper parameter types, like a `map` in this case.

I propose to either use a `key string` and a `value string` param or `map`, instead.

---

The API is like this on purpose to allow for convenient passing of multiple key / value parameters, something not possible with any of the solutions you propose. 

However, after I remove the `retries` annotation the need for setting multiple annotations in this case goes away, so I will simply remove this function.